examples/wasi.rb

```diff
@@ -0,0 +1,19 @@
+require "wasmtime"
+
+engine = Wasmtime::Engine.new
+mod = Wasmtime::Module.from_file(engine, "spec/fixtures/wasi-debug.wasm")
+
+linker = Wasmtime::Linker.new(engine)
+linker.define_wasi
```

Comment:
I started having second thoughts about this, and I've been thinking that it might better to have two constructors, one for the default case and one with WASI:
```rb
Wasmtime::Linker::new # default
Wasmtime::Linker::with_wasi 
```

I can't of a reason in which you'd need to call `define_wasi` at a later point in time after creating the linker (or maybe we discussed this and I just don't recall)

Reply:
> What's your reasoning for preferring a separate constructor?

WASI is so special and so commonly used that it seems to me that having two constructors is justified. It also feels way more solid than having to call a second method to define something that could be bundled from the start, reducing the risk of not setting up a WASI based runtime correctly. 

> By using a dedicated constructor, we're essentially making 2 types of linkers: WASI and non-WASI. That would prevent another similar concept from being used together with WASI

I see it as if you're creating a linker that automatically bundles WASI imports; but I'm not convinced that this addition prevents adding functionality in the future. I think that if we were optimizing for the extensibility use-case, an argument could be made that we shouldn't even be defining `define_wasi` in the Linker itself.